src/sema/variables.rs

```diff
@@ -427,66 +430,101 @@ pub fn var_decl(
 /// For accessor functions, create the parameter list and the return expression
 fn collect_parameters<'a>(
     ty: &'a Type,
+    symtable: &mut Symtable,
     params: &mut Vec<Parameter>,
     expr: &mut Expression,
+    ns: &mut Namespace,
 ) -> &'a Type {
     match ty {
         Type::Mapping(key, value) => {
             let map = (*expr).clone();
+            let arg_name = pt::Identifier {
+                loc: pt::Loc::Implicit,
+                name: format!("key{}", params.len()),
+            };
+            let arg_ty = key.as_ref().clone();
+
+            let arg_no = symtable
+                .add(
+                    &arg_name,
+                    arg_ty.clone(),
+                    ns,
+                    VariableInitializer::Solidity(None),
+                    VariableUsage::Parameter,
+                    None,
+                )
+                .unwrap();
+
+            symtable.arguments.push(Some(arg_no));
 
             *expr = Expression::Subscript(
                 pt::Loc::Implicit,
                 ty.storage_array_elem(),
                 Type::StorageRef(false, Box::new(ty.clone())),
                 Box::new(map),
-                Box::new(Expression::FunctionArg(
-                    pt::Loc::Implicit,
-                    key.as_ref().clone(),
-                    params.len(),
-                )),
+                Box::new(Expression::Variable(pt::Loc::Implicit, arg_ty, arg_no)),
             );
 
             params.push(Parameter {
-                id: None,
+                id: Some(arg_name),
                 loc: pt::Loc::Implicit,
                 ty: key.as_ref().clone(),
                 ty_loc: None,
                 indexed: false,
                 readonly: false,
             });
 
-            collect_parameters(value, params, expr)
+            collect_parameters(value, symtable, params, expr, ns)
         }
         Type::Array(elem_ty, dims) => {
             let mut ty = Type::StorageRef(false, Box::new(ty.clone()));
             for _ in 0..dims.len() {
                 let map = (*expr).clone();
 
+                let arg_name = pt::Identifier {
+                    loc: pt::Loc::Implicit,
+                    name: format!("index{}", params.len()),
```

Comment:
How do you make sure this name is not already used in the Symbol table? I believe that `index0` is a valid symbol for variables in Solidity.

Reply:
The only symbols in the symbol table are the parameters to the function, so there can't be a conflict.

Reply:
Don't we use the symbol table to add all variables of a function?

Reply:
Yes, we do.

This code creates an accessor function.

Reply:
That is an entirely synthetic function.

Reply:
I've changed it again. Now the parameters have no names, which probably makes more sense than coming up with synthetic useless names.